data/evidence/allegation.go

```diff
@@ -45,6 +46,32 @@ type AllegationTracker struct {
 	Requests map[string]bool
 }
 
+func (es *EvidenceStore) CleanTracker() {
+	at, err := es.GetAllegationTracker()
+	if err != nil {
+		return
+	}
+	requestIdList := make([]string, len(at.Requests))
+	for k := range at.Requests {
+		requestIdList = append(requestIdList, k)
+	}
+	sort.Strings(requestIdList)
+	countMap := make(map[string]bool)
+	for _, r := range requestIdList {
```

Comment:
Can you explain this loop? I see that you are deleting request ID's from your allegation tracker if they exist in the store. If they don't exist in the store then you save it to a count map using its address. If the request has an address already in the count map you delete it from the store. Is the allegation request already in the store at this point?

Reply:
yes, Thats exactly what its for . 
The motive is to clean , two requests with the same malicious address so that the the tracker at any point of time has oa max 1 request id per valdiator